internal/watcher/watcher_test.go

```diff
@@ -0,0 +1,140 @@
+package watcher
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAllInDirectory(t *testing.T) {
+	expectedDirCount := 2
+	expectedFilesCount := 0
+	ignoredExtensions := []string{".swp"}
+
+	baseDir, err := ioutil.TempDir("", "watcher_test1")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir: %v", err)
+	}
+
+	defer os.RemoveAll(baseDir)
+
+	subDir, err := ioutil.TempDir(baseDir, "watcher_test2")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir: %v", err)
+	}
+
+	for i := 0; i < 10; i++ {
+		file := filepath.Join(baseDir, fmt.Sprintf("test_file_%d", i))
+		if err = ioutil.WriteFile(file, []byte(":octopus:"), 0600); err != nil {
```

Comment:
I suggest using the default operating system's permission for the file creation.